core/src/test/java/org/apache/iceberg/TestPartitioning.java

```diff
@@ -354,6 +354,19 @@ public void testGroupingKeyTypeWithEvolvedUnpartitionedSpec() {
     Assert.assertEquals("Types must match", expectedType, actualType);
   }
 
+  @Test
+  public void testGroupingKeyTypeWithProjectedSchema() {
+    TestTables.TestTable table =
+        TestTables.create(tableDir, "test", SCHEMA, BY_CATEGORY_DATA_SPEC, V1_FORMAT_VERSION);
+
+    Schema projectedSchema = table.schema().select("id", "data");
```

Comment:
Ah, so is the idea to use the pushed projection to limit the fields used? I think that's a great idea!

Reply:
Correct. Otherwise, we will break Spark because we can report non-projected columns.